test/hotspot/jtreg/compiler/arguments/TestProfileCasts.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8358781
+ * @summary Regression test for -XX:-TypeProfileCasts crash
+ * @requires vm.debug
+ * @run main/othervm -XX:-TypeProfileCasts compiler.arguments.TestProfileCasts
+ */
+package compiler.arguments;
+
+public class TestProfileCasts {
+    static class Foo {
+    }
+
+    private static void test(Object o) {
+        if (o instanceof Foo) {
+        }
+    }
+
+    public static void main(String[] args) {
+        for (int i = 0; i < 100_000; i++) {
```

Comment:
Do you really need 100'000 iterations to get it to compile or can you reduce it a bit?